mantidproject/mantid
# Move away from deprecated nxs functions

[NeXus v4.4 marked `NXgetnextattr` as deprecated](https://github.com/nexusformat/code/blob/v4.4.3/include/napi.h.in#L635). This PR gets around those warnings while keeping support for NeXus v4.3.

**To test:**

Try building with nexus v4.4 and see that the warnings went away. The build servers use v4.3 and should all continue to work.

_There is no associated issue_

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

Rather than `NXgetnextattr` use `NXgetnextattra` and error
out if an array is encountered.
